### PR TITLE
fix(core): Updates dpop check for connect

### PR DESCRIPTION
**Pull request by dmihalcik-virtru**
### Proposed Changes

- Threads through Dpop token and  REST service path from gRPC gateway
  - This is needed when DPoP is enabled (required) OR when the IdP requests it

### Checklist

- [ ] I have added or updated unit tests
- [ ] I have added or updated integration tests (if appropriate)
- [ ] I have added or updated documentation

### CODE REVIEW

File: service/internal/server/server.go [hunk 1]
```diff
@@ -165,7 +167,31 @@ func NewOpenTDFServer(config Config, logger *logger.Logger) (*OpenTDFServer, err
...
+		),
+		runtime.WithMetadata(func(ctx context.Context, r *http.Request) metadata.MD {
+			md := make(map[string]string)
+			if method, ok := runtime.RPCMethod(ctx); ok {
+				md["method"] = method // /grpc.gateway.examples.internal.proto.examplepb.LoginService/Login
+			}
+			if pattern, ok := runtime.HTTPPathPattern(ctx); ok {
+				md["pattern"] = pattern // /v1/example/login
+			}
+			logger.Info("DPOP METADATATTATA", slog.Any("metadata", md))
```
Comment (elizabethhealy) [inline service/internal/server/server.go:191]: nit: maybe move this to trace

**Review by pflynn-virtru — DISMISSED** [review]

File: service/internal/server/server.go [hunk 2]
```diff
@@ -165,7 +167,31 @@ func NewOpenTDFServer(config Config, logger *logger.Logger) (*OpenTDFServer, err
...
 	// GRPC Gateway Mux
-	grpcGatewayMux := runtime.NewServeMux()
+	grpcGatewayMux := runtime.NewServeMux(
+		runtime.WithIncomingHeaderMatcher(
+			func(key string) (string, bool) {
+				if k, ok := runtime.DefaultHeaderMatcher(key); ok {
+					return k, true
+				}
+				if textproto.CanonicalMIMEHeaderKey(key) == "Dpop" {
+					logger.Info("DPOP KEY FOUUUUND")
```
Comment (elizabethhealy) [inline service/internal/server/server.go:177]: nit: same here

**Comment by strantalis** [comment]
@dmihalcik-virtru What broke with connect?  Did our x-tests miss something?

File: service/internal/server/server.go (hunk 1 above)
Comment (dmihalcik-virtru) [reply service/internal/server/server.go]: sorry meant to remove

**Comment by dmihalcik-virtru** [comment]
> @dmihalcik-virtru What broke with connect? Did our x-tests miss something?

XTests don't currently use dpop. The xtests all use client secret auth, which did not support DPoP on keycloak [until recently](https://www.keycloak.org/docs/latest/release_notes/index.html#dpop-improvements).

**Review by elizabethhealy — APPROVED** [review]
looks like the lint is failing

**Review by elizabethhealy — APPROVED** [review]

**Pull request merged**